### DISCUSSION TITLE
How can we limit package publishing?

### DISCUSSION THREAD

**Discussion by mrmckeb** [community/community#25938] [Actions]
<p>From what I read:</p>
<ul>
<li>anyone with write permissions for a repository can publish a new package to that repository</li>
<li>anyone with write permissions can publish a new package version</li>
</ul>
<p>This is fine for experienced teams, but we can see how this could go wrong when people aren’t experienced with packages and we’d like to limit publishing to certain users for the moment - and eventually, limit publishing to automation through actions.</p>
<p>What advice/suggestions are there for limiting publishing? Is this on the roadmap?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/mrmckeb">@mrmckeb</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>Typically there are two kinds of token for customer to publish a package.</p>
<ol>
<li>
<p>GITHUB_TOKEN.  It has <code>read:packages</code>  and  <code>write:packages</code>  scopes to the current repository by default(doc <a href="https://docs.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/using-github-packages-with-github-actions#about-github-packages-with-github-actions" rel="nofollow noopener">here</a>), it’s only used in Github Actions.</p>
</li>
<li>
<p>Personal access token. If the token doesn’t have <code>write:packages</code>  scopes, it cannot be used to publish a package. Please refer to the official <a href="https://docs.github.com/en/packages/publishing-and-managing-packages/about-github-packages#about-tokens" rel="nofollow noopener">doc</a> for more details.</p>
</li>
</ol>
<p>Hence, if the customer has write permission to a repository, he/she can use GITHUB_TOKEN in Github actions to publish a package, but limit to PAT token which doesn’t have wirte: package scope.</p>
<p>Cannot limit package publishing in this case.</p>
<p>Thanks</p>

**Comment by mrmckeb** [comment]
<p>Thanks <span class="mention">@weide-zhoo</span>,</p>
<p>This is definitely the answer I needed - but it’s unfortunate that there isn’t a little more control over package publishing right now.</p>
<p>Thanks again for your time!</p>
